webui/src/routes/connections/DefineSchema.tsx

```diff
@@ -38,14 +45,26 @@ const SchemaFormatEditor = ({
   format: 'json' | 'avro' | 'protobuf';
 }) => {
   type SchemaTypeOption = { name: string; value: string };
-  let schemaTypeOptions: SchemaTypeOption[] = [
+
+  const baseSchemaTypeOptions: SchemaTypeOption[] = [
     { name: 'Infer schema', value: 'inferred' },
     { name: capitalize(format) + ' schema', value: 'schema' },
   ];
 
-  if (format == 'json') {
-    schemaTypeOptions.push({ name: 'Unstructured JSON', value: 'unstructured' });
-  }
+  const schemaTypeOptions = useMemo(() => {
```

Comment:
Can you explain the reason for this change? In general, all formats should be supported for any connector, as the format system is responsible for deserializing the bytes after they've been read from the source. We also want to avoid introducing special cases for particular connectors in the UI.